src/Features/CSharp/Portable/ConvertToInterpolatedString/CSharpConvertRegularStringToInterpolatedStringRefactoringProvider.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.ConvertToInterpolatedString;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertToInterpolatedString
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.ConvertToInterpolatedString), Shared]
+    internal sealed class CSharpConvertRegularStringToInterpolatedStringRefactoringProvider :
+        AbstractConvertRegularStringToInterpolatedStringRefactoringProvider<LiteralExpressionSyntax>
+    {
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public CSharpConvertRegularStringToInterpolatedStringRefactoringProvider()
+        {
+        }
+
+        protected override bool IsAppropriateLiteralKind(LiteralExpressionSyntax literalExpression)
+            => literalExpression.Kind() == SyntaxKind.StringLiteralExpression;
+
+        protected override string GetTextWithoutQuotes(string text, bool isVerbatim)
+            => isVerbatim
+                ? text.Substring("@'".Length, text.Length - "@''".Length)
+                : text.Substring("'".Length, text.Length - "''".Length);
```

Comment:
This is less repetitive, and IMO still readable.

```suggestion
            => text[(isVerbatim ? "@'" : "'").Length..];
```

Reply:
also, please onl yhave this refactoring if there are no diagnotics on the expr.  and probably a good time to ensure that the string is terminated.  and pleas have test for thie case `"foo{`  (i.e. an unterminated literal).

Reply:
I agree, this is a better way of doing things.

(The code I used was copied from https://github.com/dotnet/roslyn/blob/master/src/Features/CSharp/Portable/ConvertToInterpolatedString/CSharpConvertConcatenationToInterpolatedStringRefactoringProvider.cs#L25)

Reply:
feel free to extract all of this into a helper utilities class they can both call into.